src/EventManager.ts

```diff
@@ -82,13 +89,13 @@ export default class EventManager {
      * @param  {string} eventName
      * @param  {function} handler
      */
-    clearEventHandler(eventName) {
+    clearEventHandler(eventName: string) {
         this.eventHandlerArrayMap.delete(eventName);
     }
 
     // returns an event listener with the js to native mapping
-    generateEventListener(eventName) {
-        const addListenerCallback = (payload) => {
+    generateEventListener(eventName: string) {
+        const addListenerCallback = (payload: any) => {
```

Comment:
can we make `payload` type more specific?

you may need to create a new variable `finalPayload` below to avoid type errors

Reply:
`payload` is one of the 11 event types:

> 4 ChangeEvents like `PermissionChange`
3 other events (`NotificationReceivedEvent` `OpenedEvent` `InAppMessageAction`) + 4 IAM lifecycle events

First, I should figure out how to use the event types you brought up above in the methods `addEventHandler` and `setEventHandler`.

Reply:
Done. `payload` is `Object`